otherlibs/systhreads/st_stubs.c

```diff
@@ -274,6 +278,7 @@ static caml_thread_t caml_thread_new_info(void)
   th = (caml_thread_t)caml_stat_alloc_noexc(sizeof(struct caml_thread_struct));
   if (th == NULL) return NULL;
   th->descr = Val_unit;
+  th->dls_state = Val_unit;
```

Comment:
I propose to initialize it correctly early:
```suggestion
  th->dls_state = Atom(0); /* Empty array */
```
This also takes care of the case of `caml_thread_register`.